src/sysext/sysext.c

```diff
@@ -689,186 +691,384 @@ static int work_dir_for_hierarchy(const char *hierarchy, const char *resolved_up
         return 0;
 }
 
-static int use_hierarchy_as_lowest_layer(
-                const char *hierarchy,
-                const char *maybe_resolved_upper_dir,
-                char** ret_hierarchy_as_lowest_layer) {
-        _cleanup_free_ char *resolved_hierarchy = NULL;
-        int r;
+struct overlayfs_paths {
+        char *hierarchy;
+        char *resolved_hierarchy;
+        char *resolved_upper_dir;
+
+        /* NULL if merged fs is read-only */
+        char *ofs_upper_dir;
```

Comment:
what does "ofs_" stand for? i mean, if this is short for "overlayfs", then the structure this is in already makes that clear. What's the different to `resolved_upper_dir`?

Reply:
"ofs" was indeed for "overlayfs". I'll drop it. 

`resolved_upper_dir` may become `ofs_upper_dir`, but it may also be a part of `ofs_layers` instead if we pass `--mutable=import`. The naming is bad here, I admit. I think I'll keep `resolved_hierarchy`, rename `resolved_upper_dir` to `resolved_extension_data` (or `resolved_mutable_layer`?), drop `ofs_` prefix and possibly rename `ofs_layers` to `lower_dirs`.

Reply:
Went with `resolved_mutable_directory`.